src/app/dashboard-widgets/environment-widget/environment-widget.component.ts

```diff
@@ -3,230 +3,55 @@ import { ActivatedRoute } from '@angular/router';
 
 import { BehaviorSubject, ConnectableObservable, Observable, Subject, Subscription } from 'rxjs/Rx';
 
-import { Context, Contexts } from 'ngx-fabric8-wit';
+import { Context, Contexts, Spaces } from 'ngx-fabric8-wit';
 
-import {
-  Environment, Space, SpaceStore
-} from '../../../a-runtime-console/index';
+import { Environment as ModelEnvironment } from '../../../app/space/create/deployments/models/environment';
 
-// non-exported classes from runtime that need to be refactored
-import { DeploymentView, DeploymentViews } from '../../../a-runtime-console/index';
-import { AppDeployments, AppEnvironmentDetails, EnvironmentDeployments } from '../../../a-runtime-console/index';
-import { AbstractWatchComponent } from '../../../a-runtime-console/index';
-import { DeploymentService } from '../../../a-runtime-console/index';
-import { ServiceService } from '../../../a-runtime-console/index';
-import { DeploymentConfigService } from '../../../a-runtime-console/index';
-import { RouteService } from '../../../a-runtime-console/index';
+import {
+  ApplicationAttributesOverview,
+  DeploymentsService,
+  Space
+} from '../../../app/space/create/deployments/services/deployments.service';
 
 @Component({
   encapsulation: ViewEncapsulation.None,
   selector: 'fabric8-environment-widget',
   templateUrl: './environment-widget.component.html',
-  styleUrls: ['./environment-widget.component.less']
+  styleUrls: ['./environment-widget.component.less'],
+  providers: [DeploymentsService]
 })
-export class EnvironmentWidgetComponent extends AbstractWatchComponent  implements OnInit, OnDestroy {
+export class EnvironmentWidgetComponent implements OnInit, OnDestroy {
 
-  loading: Subject<boolean> = new BehaviorSubject(true);
+  spaceId: Observable<string>;
+  appInfos: Observable<ApplicationAttributesOverview[]>;
   contextPath: Observable<string>;
-  appsCount: number = 0;
-  contextSubscription: Subscription;
-  currentContext: Context;
-  space: ConnectableObservable<Space>;
+  applicationInfos: ApplicationAttributesOverview[] = [];
```

Comment:
Why do we need this as a separate property from `appInfos`? The existing method of doing `(apps | async)` would let you use `appInfos` in the template without needing to have the component own its own subscription as well. If you want to have fewer async pipes around then you can use an approach like this: https://toddmotto.com/angular-ngif-async-pipe